chart/templates/_helpers.yaml

```diff
@@ -216,6 +216,34 @@ If release name contains chart name it will be used as a full name.
     defaultMode: 288
 {{- end }}
 
+{{/* Helper to render git-sync credentials envs for v3/v4 */}}
+{{- define "git_sync.env_credentials" }}
+{{- $tag := .Values.images.gitSync.tag | default "v4" }}
```

Comment:
We can't rely on the tag for this - we need to support folks who have mirrored the image and changed the tag names. If we want to add conditional logic, we will need to add a new version field for gitsync.

Also, we will want a significant newsfragment for this - this is bordering a breaking change and will want to make sure we message it appropriately.

Maybe we default the version to none and add both in that case, and only the right ones if the version is set? That's non breaking?